hbase-client/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcConnection.java

```diff
@@ -114,47 +107,42 @@
 
   @Override
   protected void callTimeout(Call call) {
-    execute(eventLoop, () -> {
-      if (channel != null) {
-        channel.pipeline().fireUserEventTriggered(new CallEvent(TIMEOUT, call));
-      }
-    });
+    Channel channel = channelRef.get();
+    if (channel != null) {
+      channel.pipeline().fireUserEventTriggered(new CallEvent(TIMEOUT, call));
+    }
   }
 
   @Override
   public boolean isActive() {
-    return channel != null;
+    return channelRef.get() != null;
   }
 
   private void shutdown0() {
-    assert eventLoop.inEventLoop();
+    Channel channel = channelRef.getAndSet(null);
     if (channel != null) {
       channel.close();
-      channel = null;
     }
   }
 
   @Override
   public void shutdown() {
-    execute(eventLoop, this::shutdown0);
+    this.shutdown0();
   }
```

Comment:
Can consolidate `shutdown()` and `shutdown0()` since we aren't running on the EventLoop

Reply:
This is done.